internal/fields/validate.go

```diff
@@ -383,6 +389,86 @@ func allVersionsIncludeECS(kibanaConstraints *semver.Constraints) bool {
 	// return !kibanaConstraints.Check(lastStackVersionWithoutEcsMappings)
 }
 
+func ecsPathWithMultifieldsMatch(name string) bool {
```

Comment:
Ideally, I'd like to avoid hard coding things from `ecs@mappings` here as changing that component template will cause tests to fail somewhere else.

While we're figuring the right path of whether to adapt `ecs@mappings` or ECS itself (https://github.com/elastic/ecs/issues/2353), could we make the validation less strict and not fail if the mapping contains additional sub-fields?

Reply:
> Ideally, I'd like to avoid hard coding things from ecs@mappings here as changing that component template will cause tests to fail somewhere else.

If `ecs@mappings` doesn't differ further from ECS we should be fine, and hopefully this should be the case.

> While we're figuring the right path of whether to adapt `ecs@mappings` or ECS itself ([elastic/ecs#2353](https://github.com/elastic/ecs/issues/2353)), could we make the validation less strict and not fail if the mapping contains additional sub-fields?

The problem is that if we relax a constraint it is difficult to put it back later. This is why I would keep this to the minimum we know.